core/src/main/scala/com/nvidia/spark/rapids/tool/tuning/AutoTuner.scala

```diff
@@ -1712,16 +1712,12 @@ abstract class AutoTuner(
   private def calculatePinnedMemorySize(numExecutorCores: Int,
                                         hostOffHeapLimitSizeMB: Long): Long = {
     // Use new formula only for onPrem platform
-    if (!platform.isPlatformCSP) {
-      // Get GPU batch size in MB
-      val gpuBatchSizeMB = StringUtils.convertToMB(
-        tuningConfigs.getEntry("BATCH_SIZE_BYTES").getDefault, Some(ByteUnit.BYTE))
-      // Calculate 2 * executor cores * GPU batch size
-      val coresBatchSizeMB = 2L * numExecutorCores * gpuBatchSizeMB
-      // Calculate 1/4 * host.offHeapLimit.Size
-      val quarterHostOffHeapLimitMB = hostOffHeapLimitSizeMB / 4
+    if (!platform.isPlatformCSP && isOffHeapLimitEnabled) {
```

Comment:
nit: It might be better to rename isOffHeapLimitEnabled as isOffHeapLimitUserEnabled - to indicate that this value is set only from user-enforced.